app/main/views/users.py

```diff
@@ -293,7 +294,12 @@ def email_has_valid_buyer_domain():
     if not email_address:
         abort(400, "'email_address' is a required parameter")
 
-    return jsonify(valid=is_valid_buyer_email(email_address))
+    domain_ok = buyer_email_domain_approved(BuyerEmailDomain.query.all(), email_address.split('@')[-1])
+    if not domain_ok:
+        # If the data migration hasn't happened yet, check the .txt file
```

Comment:
The migration runs before new code gets deployed, so I don't think we should need this? i.e. the reading of the text file will definitely have happened by the time this code is live.

And if not I think we can take the risk of a buyer user trying to sign up in the few seconds it takes to run the migration, failing once, and then succeeding the next time they try it.

Reply:
Sorry for the confusion - this branch with the data migration is just 1 commit, the code changes are in https://github.com/alphagov/digitalmarketplace-api/pull/674 and will be merged separately.

Reply:
So the plan is to remove this bit once this final migration is run?  Could we do the migration before merging the code changes and avoid the interim bit?  Maybe too late to suggest that at this point?

Reply:
Yeah, I guess I'm quite cautious when it comes to data migrations that we can't easily undo. I guess the steps are currently: 
1) release schema migration 
2) release new feature code 
3) run data migration 
4) use DB for adding/checking new domains but use text file for checking against old domains 
5) release (as yet unwritten) code change to remove old functionality and check against DB for everything. 

Then 3) and 4) could potentially happen in either order. This way if the release pipeline gets blocked for whatever reason we're still ok.